test/statements/manage/ContextManagerSyntax.chpl

```diff
@@ -0,0 +1,199 @@
+//
+// TODO: Check to make sure 'const' storage kind is immutable.
+//
+
+use TrackingRecord;
+
+proc r.enterThis() {
+  writeln('entering');
+}
+
+proc r.leaveThis(in err: owned Error?) {
+  writeln('leaving');
+  if err then try! { throw err; }
+}
+
+record res {
+  proc doSomething() {
+    writeln('doing something with resource');
+  }
+}
+
+var globalRes = new res();
+
+record man {
+  var x = new r();
+
+  proc enterThis() ref {
+    writeln('proc man.enterThis() ref: res');
+    return globalRes;
+  }
+
+  proc enterThis() const ref {
+    writeln('proc man.enterThis() const ref: res');
+    return globalRes;
+  }
+
+  proc enterThis() {
+    writeln('proc man.enterThis(): res');
+    return globalRes;
+  }
+
+  proc leaveThis(in err: owned Error?) {
+    writeln('leaving');
+    if err then try! { throw err; }
+  }
+}
+
+proc doSomething() {
+  writeln('doing something');
+}
+
+proc test1() {
+  writeln('T1');
+  // Calls the value overload of 'enterThis'.
+  manage new man() do doSomething();
+}
+test1(); writeln();
+
+proc test2() {
+  writeln('T2');
+  // Calls the value overload of 'enterThis'.
+  manage new man() {
+    doSomething();
+  }
+}
+test2(); writeln();
+
+// Make sure 'manager' stays alive till end of block.
+proc test3() {
+  writeln('T3');
+  var myManager = new man();
+  manage myManager do doSomething();
+}
+test3(); writeln();
+
+// Make sure 'manager' stays alive till end of block, nested.
+proc test4() {
+  writeln('T4');
+  var myManager = new man();
+  manage myManager, new man() {
+    doSomething();
+  }
```

Comment:
Just out of curiosity, what's the order of deinitialization here? Say, one of the managers in the manager list owns another one that's declared in the outer scope. Are we able to deinit them gracefully regardless of the order in which they are declared in the `manage` statement "header"?

Reply:
The deinitialization order should be the reverse of the initialization order.